libraries/node-core-library/src/RealNodeModulePath.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fs from 'node:fs';
+import * as nodePath from 'node:path';
+
+/**
+ * Arguments used to create a function that resolves symlinked node_modules in a path
+ * @public
+ */
+export interface IRealNodeModulePathResolverOptions {
+  readlinkSync: typeof fs.readlinkSync;
+  path: Pick<typeof nodePath, 'isAbsolute' | 'normalize' | 'resolve' | 'sep'>;
```

Comment:
Make these optional so that they don't need to be passed in, then split out the setting of defaults during the constructor

Reply:
The entire options object is currently optional, and it's a little weird to override one without the other.